getsentry/sentry-dotnet
# fix: [iOS / Native AOT] - prevent use of unsupported marshalling behaviour

This PR aims to fix #3280, preventing the use of unsupported marshalling behaviour `UnwindNativeCode` for Native AOT by disabling Marshal Exceptions when the code is compiled via native AOT (as [advised here](https://github.com/getsentry/sentry-dotnet/issues/3280#issuecomment-2350654967)).



## Testing

Testing these changes is more tricky than usual since we did find a way to run this from Unit Tests.

The best way to do this is to: [on a MAC with xcode installed]
- Clone the `sentry-dotnet` repository (make sure you don't do it on a directory that is linked to iCloud)
- Git checkout `fix/#3680`
- From the open the terminal and move to the sample directory: `cd /Samples/Sentry.Samples.Maui`
- Plug a physical device via USB (if your device is not in developer mode: open xcode, it will setup the device)
- then run `dotnet publish -c Release --framework net8.0-ios /t:Run /p:_DeviceName={DeviceID}
> If you don't know the ID of your device (DeviceID):
either run `xcrun xctrace list devices`, or head to Window -> Devices and Simulators -> device id is under `Identifier`

- wait until the app opens on your device
- terminate the process
- Open the app again on your phone
- Once open press the "Throw Unhandled .NET Exception (Crash)"
- Head to Sentry, you should see the event popping up

### Known error while testing (and their workaround)
- `error : strip exited with code 13`: this is [a known issue with Xcode 16](https://github.com/xamarin/xamarin-macios/issues/19157) (now fixed on Xcode16.2 beta) If you get this error, add the following line to the .csproj of the maui project:
```Xaml
<MtouchNoSymbolStrip Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">True</MtouchNoSymbolStrip>
```
